File: Golang/Easy/118_Pascal_Triangle/solution.go
package pascaltriangle

func generate(numRows int) [][]int {
	if numRows == 1 {
		return [][]int{{1}}
	}
	if numRows == 2 {
		return [][]int{{1}, {1, 1}}
	}

	rows := generate(numRows - 1)
	lastRow := rows[len(rows)-1]

	newRow := []int{1}
	for i := 0; i < len(lastRow)-1; i++ {
		newRow = append(newRow, lastRow[i]+lastRow[i+1])
	}
	newRow = append(newRow, 1)
	rows = append(rows, newRow)
	return rows
}

File: Python/Easy/118_Pascal_Triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        if numRows == 1:
            return [[1]]
        if numRows <= 2:
            return [[1],[1,1]]

        row = Solution.generate(self, numRows-1)
        lastRow = row[-1]

        newRow = [1]
        for i in range(0, len(lastRow)-1):
            newRow.append(lastRow[i] + lastRow[i+1])
        newRow.append(1)
        row.append(newRow)
        return row